src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -478,6 +478,128 @@ class IndexManagerTests extends HyperspaceSuite with SQLHelper {
     }
   }
 
+  test("Verify index maintenance (create, refresh) works with globbing patterns.") {
```

Comment:
Can you add create & refresh index tests for multiple `*` patterns ? e.g. `table/*/m=1/*`

Reply:
thanks @sezruby , i added a test for multiple levels of `*`. 

For specialized case e.g. `/*/m=1/*`, i think it would be an overkill, because I mean if 
`spark.read.parquet("glob/*/m=1/*")` works then this will also work right, as long as we pass in the right value in `.parquet(<path>)`? WDYT?

Reply:
You mean - is it required additional changes? or not necessary as it will work? 

And could you add some tests for the other glob patterns?  like 
```
* (match 0 or more character)
? (match single character)
[ab] (character class)
[^ab] (negated character class)
[a-b] (character range)
{a,b} (alternation)
\c (escape character)
```
I think we should check if the index with these patterns will be refreshed correctly.

Reply:
> You mean - is it required additional changes? or not necessary as it will work?
> 
> And could you add some tests for the other glob patterns? like
> 
> ```
> * (match 0 or more character)
> ? (match single character)
> [ab] (character class)
> [^ab] (negated character class)
> [a-b] (character range)
> {a,b} (alternation)
> \c (escape character)
> ```
> 
> I think we should check if the index with these patterns will be refreshed correctly.

I think it's overkill to add test for every regex possible since we are not adding any regex/wildcard handling logic at all.
cc @imback82 , @pirz , @rapoth for opinions?

Reply:
I mean if spark.read.parquet("any-wildcard-pattern") works and we make sure that we are passing the same argument always, then it will work always right?

Reply:
Yeah but we just have the tests using "*" - we can add a test using all of those patterns are used. 
I just worried about handling special characters in json, and not wanted to test regex behavior.

Reply:
Thanks, I have added this https://github.com/microsoft/hyperspace/issues/284 issue for tracking json conversion tests

Reply:
Cool, I am merging this to master. Let's do a follow up PR for this discussion. Thanks!